### ISSUE TITLE
Deprecate Reverse Engines and modify Turning Engines instead.

### ISSUE THREAD

**Issue by WildLynxDev** [endless-sky/endless-sky#3715]
(in response to #3252)

My idea:
Reverse Engines are redundant.

What should be instead:
Turning engines provide reverse thrust, thrust value is function of turning value.

And more:
When shift pressed, using direction keys causes turning engines to provide lateral thrust instead of turning. 
Alternatively, use IJKL keys for lateral movement using turning engines.

Optional:
"Shift-forward" key combo engages both main engines and turning engines for marginal speed increase.
(I understand, that "vacuum drag" is needed for gameplay balance, but I still feel a little uncomfortable with it)

Optional:
AI uses turning engines's alt mode to strafe targets, especially drones, fighters, interceptors, light combat ships...

P.S.
KSP player.

**Comment by jafdy** [comment]
And this improves the gameplay experience because ______________________________ .

You haven't yet told us why this is a good thing.

**Comment by KiLEdEnNis** [comment]
what Jafdy means is that you basically just asked "do this", without actually giving us any reasons to
what will we get from this? convince us
a downside is that turning engines would become a bit confusing and possibly too good
if you want to use your idea in some plugin, though, go ahead

**Comment by Tadrix** [comment]
Making such an outfit (a set of maneuver engines that provide both turning and reverse thrust) doesn't require any code change. You can make a plugin containing it yourself and test if it's a good idea. And if it is, convince us. (Begin by looking at Syndicate and Hai engine sets, and at the reverse thruster, in game data.)

**Comment by gunqqer** [comment]
I like the idea, but I'd need to test it out to see if it's worth it. IMO reverse engiens need a rework as it is always better to get a bigger turning engine

**Comment by WildLynxDev** [comment]
Why?

1. It's just feels wrong, that they does not ... because used I to think that turning thruster is the same as lateral thruster ... KSP legacy. Could not think the other way. Kinda my personality issue.
2. Because it adds rich manoeuvring options, especially for light ships, like "circle and strafe" tactics. Inta-hit laser turrets lower value of that tactic although. Also helps to avoid missiles, a doom for light ships.

It need to to be balanced so light ships can use and abuse this change, but bigger ships have no big bonus from it, so they are clumsy in manoeuvring and rely on turrets in combat while using gun slots for guided missiles. I think high end turning engines are a bit overpowered. Shield Beatles, for example, fly like fighters with default configuration.

**Comment by Tadrix** [comment]
So you had in mind lateral movement.

Sounds interesting, but you'd need two more keys for controls (and at this stage it's a problem, seeing as assigned keys crowd the keyboard already).

**Comment by ghoulavenger** [comment]
Way I visualize it is that a turning thruster applies rotational force on the body of the ship. For it to provide reverse thrust, the thrusters would have to be rotational themselves, which is possible, but considering that not every ship is basically a cylinder very unlikely (meaning that the thrusters could actually shoot exhaust back at the ship if they weren't in a fixed position). You could theoretically run the steering thrusters in a fixed position so they would provide reverse thrust by default but that would make them significantly weaker rotational thrusters (which you would want pushing thrust against the ship directly).

A lateral thruster is indeed within the realm of possibility, but I consider them fundamentally different from reverse thrusters. They aren't actually too different from steering though, since adjustments to steering are made by apply lateral thrust at either the front or rear of the ship, and most likely at both, (back left and front right to turn left, or back right and front left to turn right etc). Granted, the way the game portions engine space, it heavily suggests that rotational thrusters are linked to the rear of the ship only, since engine space is not partitioned and considered a whole unit and engines are always at the rear of the chasis. This is confirmed by reverse thrusters taking up weapon space, which is in the forward of the ship chasis.

So there is an argument that the rotational thrusters are lateral so you should be able to strafe so to speak, but I think the actual design of the ships suggests strongly against it. It certainly seems against pushing reverse thrust against your own ship with a rotational thruster.

**Comment by tehhowch** [comment]
I mentioned working up a PR to add lateral thrust as a function of turn rate in a Steam thread a month or two ago - I won't spend time on it just yet as I have plenty of other open PRs that are ready for consideration without me going and adding even more, especially when the current focus is on finishing story.

As far as turning thrusters offering reverse thrust capability by default.. that gets a hard "no" from me. It's conceivable that some race makes a turning thrusters that acts in this manner, sure. But every race? Nope. And the current system already supports a turn thruster having reverse thrust attributes.

**Comment by Wrzlprnft** [comment]
MZ said that he might want to include more experimental changes in unstable releases, maybe something like this could be part of it.

**Comment by WildLynxDev** [comment]
How it all works:
![how_rcs_works](https://user-images.githubusercontent.com/19364322/39399594-43a87054-4b29-11e8-8388-ade5e311e954.png)

**Comment by KiLEdEnNis** [comment]
we know how such turning works, we understand physics, but that's not the problem
the main concerns are balance, and making it non-confusing in-game

**Comment by ghoulavenger** [comment]
@WildLynxDev Your image is not representative of the game mechanics, even if it is physically relevant (i.e. can work this way). There are three distinct problems with this image.

1. There are thrusters in the forward section of the ship. Since engine space is distinct from weapon space, and reverse thrusters use weapon space, we can infer that there is no set mount in your typical ship chasis for forward mounted thrusters. Sure, this could theoretically be done, since the reverse thruster "borrows" from weapon space to do so, but your typical rotational thruster does not.

2. By the same logic, there is also no reason to believe there are rotational thrusters on the fore of the ship that apply lateral thrust. Granted, it would make sense if there were, but they are not necessary for rotation, you could very easily use just the aft of the ship to accomplish the same thing. 

3. Rotational thrusters do not apply forward thrust. Your image suggests that they do.

We could easily fudge number 2, since it would make sense that there were more than two lateral thrusters. And because of this fudging, lateral thrust makes sense, if we didn't fudge this, lateral thrust would be impossible, you would need thrusters near the center of mass to do lateral thrust rather than rotational thrust in this case -- which could be a third engine outfit that didn't take up engine space or weapon space.

We could also potentially fudge number 3, but that leads into an issue of whether forward thrusters are required at all. Rotational thrusters can get quite powerful since they are designed with a specific ship type in mind. We could easily just put a fighter with rotational thrusters and reach astronomical speeds, or make large ships turn fast enough and still be capable of moving (which would be great on freighters that have terrible turn rates and don't really need to be swift in any other way).

Before we can fudge number 1, we would have to deprecate or change reverse thrusters to use engine space. You proposed their deprecation, but I find this unlikely because they are a niche outfit already. Changing them to use engine space is more feasible, but then they just wouldn't be used at all.

All of this brings into question, the real balance issue: the usage of engine space to define a ship chasis. Since theoretically all engines are in the aft of the ship, this image represents engines being all over the ship instead.

And that doesn't consider that some people prefer the way ships fly without reverse thrusters, since they already change how your ship behaves.

**Comment by WildLynxDev** [comment]
1. Real life reference.

The picture shows how it (usually) looks in real life.

Here's real life example, somewhat large craft like those in the game.

https://www.smashwords.com/extreader/read/126031/3/americas-space-shuttle-reaction-control-system-nasa-astronaut-training-manual-rcs-2102a

Same thrusters are used both for rotation and translation or combination of thereof.
Thrusters are controlled by control computer, which converts control inputs into timing and amount of pulses each control thruster execute. Control thrusters usually fire in series of very short pulses.
Usually, they are not located near CM, instead they are located in the "corners".
When you need pure translation, they fire in pairs, by 4, and so on, so rotational moments are cancelled and linear moments add up. Vector algebra...

2. Gameplay. 
I'm not game designer, so I cannot foresee all the complications proposed changes can cause, but here is some uses for the proposed change:

2.1 Fighters and alike, especially under AI control, can do lateral strafing runs and other stuff like dodging missiles.
2.2 Large ships can use translation mode to fine tune their vectors of motion to follow the asteroids or zero relative velocity to fix their position near the stars for fuel regeneration.

**Comment by ghoulavenger** [comment]
I'm not interested in real life references, I understand how the mechanics work. It isn't too different from twisting a pencil. If you push a pencil near one of the poles, you will actually roll the pencil, but if you put weight on the center of the pencil, it rotates. If you add force to either side of the pencil, it will rotate almost perfectly, even without resistance in the center.

We can assume that the mass of the ship is significantly more than the rotational thrust applied, so it will not move laterally with only rear mounted rotational thrusters, it will only turn, just like your finger on a pencil. This isn't hard to understand or conceptualize. We don't need real life examples to understand that reference. But any changes considered have to take gameplay mechanics into account.

If rotational thrusters USED weapon space as part of their dynamic, then they COULD have forward mounted thrusters, and therefore PROVIDE reverse thrust, but they don't. So in generics, your typical rotational thruster just isn't set up the way your image suggests it to be. It can be changed fairly easily (since reverse thrust is already an in game mechanic, but then you have balance issues to go along with it, which cannot be ignored).

**Comment by Pointedstick** [comment]
What would be the in-game benefit of this proposal?

**Comment by WildLynxDev** [comment]
> What would be the in-game benefit of this proposal?

Help to solve ever unsolved problem of "interceptors are useless and fighters die like flies" ?

**Comment by jafdy** [comment]
By making them fly within range more? You might need to clarify a little.

**Comment by Zitchas** [comment]
Having fighters and destroyers that strafe would definitely expand the AI move set a bit, which would add some welcome variety to the default "heads straight for them, shoot, go past, turn around, repeat" routine that most ships do.

I think the biggest gameplay benefit, however, would be that a fair number of *players* would enjoy the ability to strafe. I know I would. And I would quite happily allot an extra two or three buttons to control said behaviour, too.

**Comment by Pointedstick** [comment]
> Help to solve ever unsolved problem of "interceptors are useless and fighters die like flies" ?

I'm not sure it would help that at all. Strafing fighters would still be within gun range, which means they could still get shredded quickly. Fighters in gun range gain survivability with greater fighter numbers or lower turret turn rates.

This Issue is pretty clogged up with unrelated arguments and it's hard to tell what the real request actually is. I recommend opening a new issue in which you directly request what you actually want rather than first proposing an implementation of it, which tends to generate arguments.
